proto/regen/ecocredit/v1alpha2/tx.proto

```diff
@@ -72,19 +74,47 @@ message MsgCreateClass {
 
 // MsgCreateClassResponse is the Msg/CreateClass response type.
 message MsgCreateClassResponse {
-
   // class_id is the unique ID of the newly created credit class.
   string class_id = 1;
 }
 
+// MsgCreateProjectResponse is the Msg/CreateProject request type.
+message MsgCreateProject {
+  // issuer is the address of an approved issuer for the credit class through
+  // which batches will be issued. It is not required, however, that this same
+  // issuer issue all batches for a project.
+  string issuer = 1;
+
+  // class_id is the unique ID of the class within which the project is created.
+  string class_id = 2;
+
+  // metadata is any arbitrary metadata attached to the project.
+  bytes metadata = 3;
+
+  // project_location is the location of the project backing the credits in this
+  // batch. It is a string of the form
+  // <country-code>[-<sub-national-code>[ <postal-code>]], with the first two
+  // fields conforming to ISO 3166-2, and postal-code being up to 64
+  // alphanumeric characters. country-code is required, while sub-national-code
+  // and postal-code can be added for increasing precision.
+  string project_location = 4;
+}
+
+// MsgCreateProjectResponse is the Msg/CreateProject response type.
+message MsgCreateProjectResponse {
+
+  // project_id is the ID of the newly created project.
+  string project_id = 1;
+}
+
 // MsgCreateBatch is the Msg/CreateBatch request type.
 message MsgCreateBatch {
 
   // issuer is the address of the batch issuer.
   string issuer = 1;
 
-  // class_id is the unique ID of the class.
-  string class_id = 2;
+  // project_id is the unique ID of the project this batch belongs to.
+  string project_id = 2;
```

Comment:
Similar to what I said in the `types.proto` file- it may be nice to keep `class_id` for easier querying.

Additionally - if not all `project_id`s are actually corresponding to on-chain entities (e.g. for Verra credit classes), then I think we still should require the `class_id` field here.

Reply:
I agree it would make it easier for querying.

Reply:
> I agree it would make it easier for querying.

i recommend queries at the credit class and methodology level. what are use-cases scenarios of the query function?

Reply:
This is a type to create a new batch. It seams redundant to provide a class here - could be error prone. Each project is assigned to only one class. So having a project id we have its associated class id.

We can add a query method by class id (to return batches by class id) separately.

Reply:
Hi Robert. A carbon credit project can be attached to multiple credit classes. This is true for Verra credits purchased OTC that they link to multiple methodologies. In the case of Regen produced credits, they are usually formatted as primary indicator (credit class) plus cobenefits. However, there is not always a 1:1 project:credit class relationship in the industry, so the credit class reference needs to be coded to accept multiple credit class types per project.

Reply:
While this is true with other registries & how they manage projects, since we don't have the concept of a "registry" as an onchain entity at the moment, its not really possible to have projects defined on a 'per registry basis', which is why in this initial PR, we're defining projects on a 'per credit class' basis.

If we go with @aaronc's proposal of allowing project_id's to be manually set by the creator of a project, then this should work out fine. Different credit classes could have their own issuer list and management process for creating projects under that credit class, and when those projects are created on-chain they could still point to the same "Verra" project, by either having the same metadata, or the same `V-456` project_id, or both.

Reply:
In this model a project can only be associated with one class. If we want to change that then this will need a redesign